moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -777,12 +783,17 @@ bool ServoCalcs::applyJointUpdate(const Eigen::ArrayXd& delta_theta, sensor_msgs
 
   smoother_->doSmoothing(joint_state.position);
 
-  for (std::size_t i = 0; i < joint_state.position.size(); ++i)
-  {
-    // Calculate joint velocity
-    joint_state.velocity[i] =
-        (joint_state.position.at(i) - original_joint_state_.position.at(i)) / parameters_->publish_period;
-  }
+  // Lambda that calculates velocity using central difference.
+  // (q(t + dt) - q(t - dt)) / ( 2 * dt )
+  auto compute_velocity = [&](const double& next_pos, const double& previous_pos) {
```

Comment:
nit, but there's no need to pass basic types like double by reference here

Reply:
yep, it actually hurts performance (a bit...)

Reply:
Would it not be faster than pass by value which makes a copy ?

Reply:
it's a complex question but our rule of thumb is what Sebastian commented

https://blog.bytellect.com/software-development/cplusplus/which-is-faster-in-cplusplus-call-by-value-call-by-reference-or-call-by-pointer/

Reply:
Oh nvm, it is slower for basic types, due to unnecessary extra memory access.

Reply:
tl;dr the address for basic types basically uses the same memory as the payload itself, so pass-by-reference only pays off when the data types are nontrivial

Reply:
Might also be good to restrict the variables captured by the lambda to just the 3 arrays it access?

Reply:
I don't think that will matter from a performance standpoint